transfers/reingest.py

```diff
@@ -0,0 +1,509 @@
+        LOGGER.error("Reingest failed with server error %s, returning.",
+                     errors.error_lookup(reingest_aip))
+        return False, "Error calling reingest_aip."
+
+    reingest_uuid = reingest_aip["reingest_uuid"]
+    LOGGER.info("Reingest UUID to work with %s", reingest_uuid)
+
+    LOGGER.info("Checking status of %s once in transfer queue", reingest_uuid)
+
+    # We need to make sure that the approval is synchronized with the request
+    # to reingest. Occasionally a reingest will get stopped at approval when
+    # it doesn't need to if we just wait a little longer or try again.
+    for retry in range(approval_retries):
+        transfer = None
+        while not isinstance(transfer, dict):
+            if latency is not None and isinstance(latency, float):
+                time.sleep(latency)   # ~latency between call and AM actioning.
+            transfer = amclient.get_transfer_status(reingest_uuid)
+
+        LOGGER.info("Attempting to approve transfer following reingest init")
+
+        if transfer["status"] == "USER_INPUT":
+            transfer_directory = transfer["directory"]
+            LOGGER.info("Approving reingest automatically. Directory to "
+                        "approve %s", transfer_directory)
+            message = amclient.approve_transfer(transfer_directory)
+            if 'Approval successful.' in message['message']:
+                LOGGER.info("Approval successful, returning True")
+                return True, message['uuid']
+            return False, "Error approving transfer."
+    return False, "Error retrieving transfer status."
+
+
+def get_am_client(config):
+    """Return an AM Client Object to with throughout the rest
+    of the script.
+    """
+    connection = config['connection']
+    return amclienthelper.AMClient(
+        ss_url=connection['ss_url'],
+        ss_user_name=connection['ss_user_name'],
+        ss_api_key=connection['ss_api_key'],
+        am_url=connection['am_url'],
+        am_user_name=connection['am_user_name'],
+        am_api_key=connection['am_api_key'],
+        output_mode=connection['output_mode'])
+
+
+def manage_process(config_file, remove=False):
+    """Manage the reingest process using a process lock. If the PID.lck file
+    exists then inform the user. If called upon, remove the PID.
+    """
+    pid_file = config_file['process']['pid']
+    if remove:
+        if os.path.isfile(pid_file):
+            LOGGER.info("Removing PID for current process.")
+            os.remove(pid_file)
+            return
+        return False
+    try:
+        pidf = os.fdopen(
+            os.open(pid_file, os.O_CREAT | os.O_EXCL | os.O_RDWR), 'r+')
+    except BaseException:
+        LOGGER.info('This script is already running. To override this '
+                    'behavior and start a new run, remove %s', pid_file)
+        return
+    pid = os.getpid()
+    pidf.write(str(pid))
+    pidf.close()
+    return pid_file
```

Comment:
This is a good catch. I think I've captured a bit more in the docstring and made changes to the function to improve its logic. It might take an iteration or two to get right but let me know.